design/specs/listing-views.md

```diff
@@ -1,27 +1,109 @@
+## Context
 
-# Prototype
-[Prototype for Listing and Finding Views](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?page-id=5343%3A76906&node-id=5343%3A76907&viewport=338%2C48%2C0.72&scaling=contain&starting-point-node-id=5343%3A76907)
+Database objects are defined objects in a database that store or reference data. In the context of Mathesar, two of those objects are listed and accessible to users as `Tables` and `Views`. Together, these objects are at the core of Mathesar's user experience. One part of that experience is finding and opening these objects and differentiating both.
+
+## Prototype
+
+### Link
+
+[Prototype](https://mathesar-prototype.netlify.app/)
+
+### Videos
+
+- Scenario 1: [Video](https://www.loom.com/share/179070149f00419e9125bd8fba675aa4)
+- Scenario 2: [Video](https://www.loom.com/share/e32500bc744246e5b926852231577fd7)
+- Scenario 3: [Video](https://www.loom.com/share/d553e7edbb28481b8aed02b055270154)
+- Scenario 4: [Video](https://www.loom.com/share/91a20c4770ef4439860425a1bcd84db6)
+
+## User Experience
+
+### Listing of Views and Searching by Name
+
+With the introduction of `Views`, the user interface needs to be updated to retrieve and identify all objects easily. `Views` in the same manner as `Tables`, have names that allow users to identify them or search them.
+
+Initially, users will only be able to search views by name. However, if a user hasn't encountered a particular object very often, relying on a name alone might not be sufficient to ensure a good user experience.
 
-# User Experience
 ## Scenarios
 
-### User sees a list of all views and can filter or sort the list items
-A user can see a list of all views for a schema in the sidebar area. They can filter the list using the search input or sort them by alphabetical order or last modified. Last modified is the default sorting for the list of views.
+### Scenario 1: User browses through a list of all available views in a currently open schema
+
+#### Scenario 1a: From the schema details view
+
+- User navigates to the schema details page by navigating directly to a URL that points to this page or by clicking on the schema link at the top navigation bar.
+- User sees a list of all tables and views for the current schema.
+  - By default, the list is sorted by the last accessed object.
+- Above a certain number of objects, the list will be divided into pages. The user will then have to navigate the pages to access the entire list.
+
+#### Scenario 1b: From the schema explorer view's sidebar
+
+- User navigates to the schema explorer view by opening any object in the details page or navigating to an object URL.
+- User sees a list of all tables and views for the current schema.
+  - The list shows both object types by default but can be filtered to only show tables or views using the filter bar at the top of the sidebar.
+
+### Scenario 2: User encounters an error in loading a list of views for a currently open schema
```

Comment:
We need UX to show the error on the sidebar as well.

There is a possibility that the entries fail to load on the sidebar while the schema detail page loads successfully.